cfdm/read_write/netcdf/netcdfread.py

```diff
@@ -11273,6 +11288,9 @@ def _cache_data_elements(self, data, ncvar):
 
             elements[index] = value
 
+        # Cache the cached data elements for this variable
+        g["cached_data_elements"][ncvar] = elements
```

Comment:
Why do we run also run `data._set_cached_elements(elements)` in the line after this? It that (still) necessary?

Reply:
Hi Sadie - `data._set_cached_elements(elements)` caches the elements on the `Data` object, but the way the code is at the moment, all metadata netCDF variables get read twice (once as a potential field, and once as metadata), so we want to cache the elements at the `netcdfread.py` level, too, so that the second read doesn't have to go back to the file.

Reply:
Oh, hang on! But elements aren't cached for Field constructs, so there should be no double-reading at all ...  In that case maybe this issue is not an issue at all! I'll have a think and a dig ....